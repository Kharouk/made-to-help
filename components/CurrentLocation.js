import React from 'react';
import { Marker } from 'react-native-maps';
import { Location, Permissions } from 'expo';

export default class CurrentLocation extends React.Component {
  state = {
    longitude: 0,
    latitude: 0,
  };

  componentWillMount() {
    this._getLocationAsync();
  }

  _getLocationAsync = async () => {
    await Permissions.askAsync(Permissions.LOCATION);

    const location = await Location.getCurrentPositionAsync({});
    this.setState({ longitude: location.coords.longitude, latitude: location.coords.latitude });
  };

  render() {
    return (
      <Marker
        coordinate={{ latitude: this.state.latitude, longitude: this.state.longitude }}
        image={require('../assets/images/user.png')}
      />
    );
  }
}
