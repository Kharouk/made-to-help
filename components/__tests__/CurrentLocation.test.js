import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import CurrentLocation from '../CurrentLocation';

Enzyme.configure({ adapter: new Adapter() });

test('Renders CurrentLocation component', () => {
  const snapshot = shallow(<CurrentLocation />);
  expect(snapshot).toMatchSnapshot();
});

describe('Current Location', () => {
  it('shallow renders', () => {
    const wrapper = shallow(<CurrentLocation />);
    expect(wrapper.state('longitude')).toBe(0);
  });

  it('test location method changes latitude', () => {
    const wrapper = shallow(<CurrentLocation />);
    const instanceComponent = wrapper.instance();
    instanceComponent._getLocationAsync = () => {
      wrapper.setState({ longitude: 51 });
      wrapper.setState({ latitude: -0.03 });
    };
    instanceComponent.componentWillMount();
    expect(wrapper.state('longitude')).toBe(51);
    expect(wrapper.state('latitude')).toBe(-0.03);
  });
  
  it('test location async', () => {
    const wrapper = shallow(<CurrentLocation />)
    const location = {coords:{latitude: -0.03, longitude: 50}}
    wrapper.setState({ longitude: location.coords.longitude})
    wrapper.setState({ latitude: location.coords.latitude})
    expect(wrapper.state('latitude')).toBe(-0.03)
    expect(wrapper.state('longitude')).toBe(50)
  })
})
