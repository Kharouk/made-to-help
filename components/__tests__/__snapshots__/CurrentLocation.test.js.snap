// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders CurrentLocation component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CurrentLocation />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "coordinate": Object {
        "latitude": 0,
        "longitude": 0,
      },
      "image": 1,
      "stopPropagation": false,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "coordinate": Object {
          "latitude": 0,
          "longitude": 0,
        },
        "image": 1,
        "stopPropagation": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
